pages/lessons/projects/6.mdx

```diff
@@ -0,0 +1,796 @@
+
+Now that we have our house built, we need to register it so as to guarantee the
+property rights. We can think of this as deploying our contract to a blockchain,
+where our token will be fully functioning for every address that wants to
+interact with it.
+
+We are going to need a small amount of Goerli ETH to be able to pay for the
+transaction to deploy our contract. If you already have some Test ETH in Goerli,
+you can continue, if you don't you can always ask for some in the many faucets
+that are online. Some options are: [Alchemy](https://goerlifaucet.com/),
+[Goerli Faucet](https://goerlifaucet.org/) or
+[Paradigm](https://faucet.paradigm.xyz/), but you can find more if you search
+for Goerli Faucet in any search engine.
+
+We are going to create an `.env` file, where we will store our _secrets_. And
+when we mean _secrets_, we are specially referring to the private key to your
+wallet. Never, ever share this with anyone, not even us. Anyone who gets hold of
+your keys, can drain your wallet.
+
+This is why we strongly advice to use a separate wallet for our development,
+that should not hold anything with real value in it. It's not the first time,
+even for experienced developers, that we accidentally upload our `.env` file to
+GitHub, and once it's on the internet we can consider it to be public.
+
+So, with that important notice out of the way, we create a new file in our
+project's root folder called `.env` and we should fill it with this:
+
+```env
+RAW_PRIVATE_KEY=REPLACE_THIS_VALUE_WITH_YOUR_PRIVATE_KEY
+ETH_RPC_URL=https://rpc.ankr.com/eth_goerli
+CHAIN=goerli
+ETHERSCAN_API_KEY=REPLACE_WITH_ETHERSCAN_API_KEY
+```
+
+Remember to replace the `REPLACE_THIS_VALUE_WITH_YOUR_PRIVATE_KEY` with your
+development wallet's private.
+In the spirit of Open Source Software (OSS) we recommend you to also replace the
+`REPLACE_WITH_ETHERSCAN_API_KEY` with your etherscan API KEY so we can verify
+our contract.
+
+Overview of our values:
+
+- **RAW_PRIVATE_KEY**: Here we shall paste our development wallet's private key
+  after the `=` sign
+- **ETH_RPC_URL**: This is the token from our RPC provider, we are using a public
+  one but you can use your Infura, Alchemy, Quicknode or whatever provider you
+  prefer
+- **CHAIN**: The blockchain where we will be deploying out contract. We are using
+  the goerli testnet since this is not intended to be on mainnet.
+- **ETHERSCAN_API_KEY**: Here we will use our Etherscan API KEY so we can verify
+  our contract for everyone to see the deployed code and foster transparency in
+  the web3 space.
+
+Once we have this values inside the file, we can source it to have this values
+as enviroment variables before using foundry's deploy tools.
+
+There are many ways to load these values into environment variables, depending
+on which shell you use (Bash, zsh, fish...). The next command works with most of
+bash-compatible shells:
+
+```bash
+export $(grep -v '^#' .env | xargs)
+```
```

Comment:
Awesome! Some learners might be used to installing dotenv and adding the code for its use. How easy is that!

Reply:
Yeah, I don't know if I need to clarify that this gets wiped when you end your terminal session.

Reply:
I'm adding this so people know that they need to run this again if they close the terminal